--- a/Lib/site.py 2019-09-06 23:59:43.080736298 +0300
+++ b/Lib/site.py 2019-09-07 00:01:05.134165490 +0300
@@ -340,6 +340,7 @@
         else:
             sitepackages.append(prefix)
             sitepackages.append(os.path.join(prefix, "lib", "site-packages"))
+        sitepackages.insert(0, os.path.join(prefix, "lib", "sulin"))
     return sitepackages
 
 def addsitepackages(known_paths, prefixes=None):

