--- Lib/site.py.orig	2014-05-31 21:58:39.000000000 +0300
+++ Lib/site.py	2014-06-03 01:06:08.958350806 +0300
@@ -292,6 +292,7 @@ def getsitepackages():

                                         "python" + sys.version[:3],
                                         "site-packages"))
             sitepackages.append(os.path.join(prefix, "lib", "site-python"))
+            sitepackages.insert(0, os.path.join(prefix, "lib", "sulin"))
         else:
             sitepackages.append(prefix)
             sitepackages.append(os.path.join(prefix, "lib", "site-packages"))
