--- functions
+++ functions
@@ -946,24 +946,10 @@
 
 ################################################################ extraction
 
-EXTRACTORS_SUPPORTED="dpkg-deb ar"
+EXTRACTORS_SUPPORTED="ar"
 EXTRACT_DEB_TAR_OPTIONS=
 
-# Native dpkg-deb based extractors
-extract_dpkg_deb_field () {
-	local pkg field
-	pkg="$1"
-	field="$2"
 
-	dpkg-deb -f "$pkg" "$field"
-}
-
-extract_dpkg_deb_data () {
-	local pkg="$1"
-
-	dpkg-deb --fsys-tarfile "$pkg" | tar $EXTRACT_DEB_TAR_OPTIONS -xf - || error 1 FILEEXIST "Tried to extract package, but file already exists. Exit..."
-}
-
 # Raw .deb extractors
 extract_ar_deb_field () {
 	local pkg field tarball
@@ -1024,18 +1010,12 @@
 
 	if [ -n "$EXTRACTOR_OVERRIDE" ]; then
 		extractor="$EXTRACTOR_OVERRIDE"
-	elif in_path dpkg-deb; then
-		extractor="dpkg-deb"
 	else
 		extractor="ar"
 	fi
 
 	info CHOSENEXTRACTOR "Chosen extractor for .deb packages: %s" "$extractor"
 	case "$extractor" in
-	dpkg-deb)
-		extract_deb_field () { extract_dpkg_deb_field "$@"; }
-		extract_deb_data () { extract_dpkg_deb_data "$@"; }
-		;;
 	ar)
 		extract_deb_field () { extract_ar_deb_field "$@"; }
 		extract_deb_data () { extract_ar_deb_data "$@"; }
